File: framework/fit/cpp/plugin/fit_dependency/component/com_huawei_fit_hakuna_kernel_registry_shared_Application/1.0.0/cplusplus/formatter/json/Converter.cpp
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2020-2021. All rights reserved.
 * Description  : auto generate by FIT IDL
 * Author       : auto
 * Date         :
 */

#include "Converter.hpp"
#include <fit/external/util/registration.hpp>
#include <fit/external/framework/formatter/formatter_collector.hpp>

namespace Fit {
namespace Framework {
namespace Formatter {
namespace Json {
template <>
FitCode MessageToJson(ContextObj ctx, const ::fit::hakuna::kernel::registry::shared::Application &value,
    rapidjson::Writer<rapidjson::StringBuffer> &writer)
{
    writer.StartObject();

    writer.String("name");
    MessageToJson(ctx, ExtractArgToRef(value.name), writer);
    writer.String("nameVersion");
    MessageToJson(ctx, ExtractArgToRef(value.nameVersion), writer);
    writer.String("extensions");
    MessageToJson(ctx, ExtractArgToRef(value.extensions), writer);

    writer.EndObject();
    return FIT_OK;
}


template <>
FitCode JsonToMessage(ContextObj ctx, const rapidjson::Value &jsonValue,
    ::fit::hakuna::kernel::registry::shared::Application &value)
{
    if (!jsonValue.IsObject()) {
        return FIT_ERR_PARAM;
    }
    value = {};

    if (jsonValue.HasMember("name") && !jsonValue["name"].IsNull()) {
        JsonToMessage(ctx, jsonValue["name"], value.name);
    }

    if (jsonValue.HasMember("nameVersion") && !jsonValue["nameVersion"].IsNull()) {
        JsonToMessage(ctx, jsonValue["nameVersion"], value.nameVersion);
    }

    if (jsonValue.HasMember("extensions") && !jsonValue["extensions"].IsNull()) {
        JsonToMessage(ctx, jsonValue["extensions"], value.extensions);
    }

    return FIT_OK;
}
}
}
}
}

File: framework/fit/java/fit-util/src/main/java/com/huawei/fitframework/util/convert/Converter.java
/*
 * Copyright (c) Huawei Technologies Co., Ltd. 2023-2023. All rights reserved.
 */

package com.huawei.fitframework.util.convert;

import java.util.function.Function;

/**
 * 为转换提供方法。
 *
 * @author 梁济时
 * @since 2023-01-28
 */
@FunctionalInterface
public interface Converter extends Function<Object, Object> {
    /**
     * 将原始对象进行转换，得到目标对象。
     *
     * @param value 表示待转换的原始对象的 {@link Object}。
     * @return 表示转换后的对象的 {@link Object}。
     */
    Object convert(Object value);

    @Override
    default Object apply(Object value) {
        return this.convert(value);
    }
}
